### DISCUSSION TITLE
How to set the multilingual mode?

### DISCUSSION THREAD

**Discussion by kugarocks** [thuliteio/doks#1292] [Q&A] [answered]
Is there anything wrong with my multilingual configuration?

`hugo.toml`

```
# Multilingual
defaultContentLanguage = "en"
disableLanguages = ["de", "nl"]
defaultContentLanguageInSubdir = false
```

`menus.zh-cn.toml`

```
[zh-cn]
  languageName = "中文"
  contentDir = "content/zh-cn"
  weight = 30
  [zh-cn.params]
    languageISO = "ZH"
    languageTag = "zh-CN"
    footer = '由 <a class="text-muted" href="https://thulite.io/">Thulite</a> 提供'
    alertText = '<a class="alert-link stretched-link fw-normal" href="/blog/example-post/">Doks 1.0 刚刚发布！</a>'
```

`params.toml`

```
# Multilingual
multilingualMode = true
showMissingLanguages = true

# Nav
sectionNav = ["en", "zh-cn"]
```

File Structure

```
content
├── en
│   ├── _index.md
│   ├── blog
│   │   ├── _index.md
│   │   └── example
│   │       └── index.md
│   ├── docs
│   │   ├── _index.md
│   │   ├── guides
│   │   │   ├── _index.md
│   │   │   └── example.md
│   │   ├── reference
│   │   │   ├── _index.md
│   │   │   └── example.md
│   │   └── resources.md
│   └── privacy.md
└── zh-cn
    ├── _index.md
    ├── blog
    │   ├── _index.md
    │   └── example
    │       └── index.md
    ├── docs
    │   ├── _index.md
    │   ├── guides
    │   │   ├── _index.md
    │   │   └── example.md
    │   ├── reference
    │   │   ├── _index.md
    │   │   └── example.md
    │   └── resources.md
    └── privacy.md
```

I’ve searched a lot but still can’t find any solutions. Thank you!

* Why hasn’t this page been translated yet?
* How do I remove the top-level nodes from the sidebar?

![i18n](https://github.com/user-attachments/assets/457ae4a3-c4a5-4c10-adbb-b1ba126a6d34)

**Comment by kugarocks** [comment]
I found that I need to set these two properties to all front matter.

```
---
lang: zh-cn
url: /zh-cn/blog/
---
```

But it's still untranslated.

**Reply by h-enk** [reply]
That's not necessary (or don't do that)

**Comment by h-enk** [comment, marked as answer]
I've added a guide, example repo, and demo site describing/showing how to set up and use Doks in Multilingual mode:

- [Internationalization (i18n) guide](https://getdoks.org/docs/guides/i18n/)
- [Example repo](https://github.com/thuliteio/doks-multilingual)
- [Demo site](https://profound-caramel-63661a.netlify.app/)

**Reply by kugarocks** [reply]
OH YEAH IT WORKS! Thank you so much, you just save my tomorrow! I've been stuck on this two days. I used Cursor @codebase but didn't find any solutions. I also didn't see any configurations related to module mounts and multilingual support on the Hugo official website. This is the best theme I ever saw, thanks again.

**Reply by h-enk** [reply]
Hahaha, great to hear, and you're welcome !

**Comment by h-enk** [comment]
@kugarocks Relevant/missing part in your set up is most likely [Update mounts](https://getdoks.org/docs/guides/i18n/#update-mounts)
